src/main/java/core/basesyntax/db/Storage.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.db;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitsStorage = new HashMap<>();
+
+    public static void putFruitQuantity(String fruit, int quantity) {
+        fruitsStorage.put(fruit, quantity);
```

The method `putFruitQuantity` does not validate its inputs. The task requires validation for null parameters and non-negative quantities. Quote of violated checklist items:
- "Validate and handle invalid inputs: Null parameters."
- "Quantities must be integers and >= 0; throw RuntimeException if invalid."
Also, when validation fails you should throw a RuntimeException with an informative message (see: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and include the cause [CHECKLIST ITEM #8]"). Consider adding checks like `if (fruit == null) throw new RuntimeException("Fruit name is null");` and `if (quantity < 0) throw new RuntimeException("Quantity must be >= 0: " + quantity);` so invalid data cannot be stored silently.